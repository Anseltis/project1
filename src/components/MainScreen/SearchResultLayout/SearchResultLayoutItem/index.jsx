import React from 'react'
import { Link } from 'react-router-dom'

export const SearchResultLayoutItem = ({item, onclick, dataKey}) => {
    const clickHandler  = () => onclick({ dataKey })
    return (
        <Link to={`/${item.id}`}>
        <div className='film-layout__item' onClick={clickHandler}>
        <img className='film-layout__cover' src={item.poster_path} alt=''/>
        <div className='film-layout__title-container'>
            <span className='film-layout__title'>{item.title}</span>
            <span className='film-layout__year'>{item.release_date}</span>
        </div>
         <span className='film-layout__genre'>{item.genres[0]}</span>
    </div>
            </Link>)
}
