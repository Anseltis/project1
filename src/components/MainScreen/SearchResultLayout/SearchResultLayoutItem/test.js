import React from 'react'
import renderer from 'react-test-renderer'

import { SearchResultLayoutItem } from 'components/MainScreen/SearchResultLayout/SearchResultLayoutItem/index'

const itemMock = {
  poster_path: '',
  title: '',
  release_date: '',
  genres: ['']
}

test('SearchResultLayoutItem renders properly', () => {
  const wrapper = renderer
    .create(<SearchResultLayoutItem item={itemMock} />)
    .toJSON()
  expect(wrapper).toMatchSnapshot()
})
