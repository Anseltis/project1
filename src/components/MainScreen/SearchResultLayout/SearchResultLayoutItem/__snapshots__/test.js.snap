// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultLayoutItem renders properly 1`] = `
<div
  className="film-layout__item"
  onClick={[Function]}
>
  <img
    alt=""
    className="film-layout__cover"
    src=""
  />
  <div
    className="film-layout__title-container"
  >
    <span
      className="film-layout__title"
    >
      
    </span>
    <span
      className="film-layout__year"
    >
      
    </span>
  </div>
  <span
    className="film-layout__genre"
  >
    
  </span>
</div>
`;
